source/docs/docs/identity-and-access-management/tenancy.md

```diff
@@ -0,0 +1,67 @@
+# Tenancy
+
+## Scope
+
+Tenancy is the concept of providing isolated access to resources. In the context of the Infratographer ecosystem, we are talking about the isolation of data and the ability to provide a single application to multiple tenants. The scope of this document is to provide a shared understanding and a working model for developing multi-tenant applications within the Infratographer ecosystem. It is outside the scope to discuss multi-tenancy in other systems beyond Infratographer except where drawing comparison helps to clarify our intent.
+
+## Principles of Tenancy
+
+### Tenant Trees are Isolated
+
+Tenants provide isolated access to data. Access to data within a tenant tree needs to be explicitly granted.
+
+#### Rationale
+
+Tenant isolation limits access privileges to data and control over resources managed by the ecosystem.
+
+#### Implications
+
+Access to a parent tenant is not implied by granting access to a child tenant and access to one tenant does not imply access to other unrelated tenant trees. [Access granted to a parent tenant *does* imply granting access to child tenants](#tenant-permissions-are-inherited)
+
+### Tenant permissions are inherited
+
+Access granted on a tenant will be inherited by all subtenants.
+
+#### Rationale
+
+Inheriting access rights from the parent is more intuitive and fits better into existing authorization paradigms.
+
+#### Implication
+
+When access is granted to a tenant, that access will be granted to all subtenants of the tenant at the same level. Access cannot be limited to only one level unless the tenant has no children. In some cases, a flat tenant structure may be better serve isolation requirements.
```

Comment:
In principle, this statement is true, having access to a resource within a tenant should give you access to that same resource in child tenants within that tree. However, what is a resource in this case?

This is where having an authorization model definition that works with the tenancy model would come in handy. Should it be part of this document? I'm not sure, but we need to relate the two concepts in order to give a cohesive view of the world.

In order to do fine-grained authorization, the idea was that a permission would be `<resource type> + <action>`. A set of permissions form a "role", and a "role grant" would allow a subject or principal access to the permissions defined in that role at a tenant level. Roles grants are inherited, which means that you'd get access to the same permissions within children in the tree.

There are a lot more caveats and details to this model, but for the purpose of tenancy, this is what's relevant to know.

While a flat tenant structure may serve isolation requirements in a more straight-forward manner. Very early in a multi-tenant system's development one figures out that the flat model falls short since it disallows and/or complicate use-cases such as "reseller", "contractors", and even "customer support". (I can elaborate on these if needed)

I'd heavily advise against a flat structure for the sake of simplicity, hierarchical tenancy is a tried and proven model which we shouldn't dismiss for short-term benefits.

Reply:
Maybe there is a better way to phase this implication.  The idea I was trying to express here is that tenants are hierarchical, but users of the tenant structure may adapt a flatter model to meet isolation requirements where inherited roles are not desirable.